client/state/data-layer/wpcom/read/related/sites/index.js

```diff
@@ -0,0 +1,46 @@
+import { decodeEntities } from 'calypso/lib/formatting';
+import { registerHandlers } from 'calypso/state/data-layer/handler-registry';
+import { http } from 'calypso/state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'calypso/state/data-layer/wpcom-http/utils';
+import { READER_RELATED_SITES_REQUEST } from 'calypso/state/reader/action-types';
+import { receiveRelatedSites } from 'calypso/state/reader/related-sites/actions';
+
+const noop = () => {};
+
+export const requestRelatedSites = ( action ) => {
+	const { tag = '', number = 10, offset = 0 } = action.payload;
+	return http( {
+		method: 'GET',
+		path: `/read/tags/${ tag }/cards`,
+		query: { number, offset, posts_per_site: 0 },
+		apiNamespace: 'wpcom/v2',
+		onSuccess: action,
+		onFailure: action,
+	} );
+};
+
+export const fromApi = ( { cards } ) => {
+	const sites = cards
+		.filter( ( card ) => card.type === 'post' )
+		.map( ( card ) => ( {
```

Comment:
hmm since cards represent posts, not sites, would there be duplicates that need to be filtered out. e.g. if a site is spamming posts could we get multiple for the same site?